doc/concepts/configuration/configuration_credentials.rst

```diff
@@ -0,0 +1,121 @@
+
+This topic describes how to create users and grant them the specified privileges in the :ref:`credentials <configuration_reference_credentials>` section of a YAML configuration.
+This might be used to create specific users used in communications between Tarantool instances.
+For example, such users can be created to maintain :ref:`replication <replication-master_replica_configuring_credentials>` and sharding in a Tarantool cluster.
+
+
+.. _configuration_credentials_managing_users_roles:
+
+Managing users and roles
+------------------------
+
+.. _configuration_credentials_managing_users_roles_creating_user:
+
+Creating a user
+~~~~~~~~~~~~~~~
+
+You can create new or configure credentials of the existing users in the :ref:`credentials.users <configuration_reference_credentials_users>` section.
+
+In the example below, a ``dbadmin`` user without a password is created:
+
+.. code-block:: yaml
+
+    credentials:
+      users:
+        dbadmin: {}
+
+To set a password, use the :ref:`credentials.users.\<username\>.password <configuration_reference_credentials_users_name_password>` option:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/credentials/config.yaml
+    :language: yaml
+    :start-at: credentials:
+    :end-at: T0p_Secret
+    :dedent:
+
+.. _configuration_credentials_managing_users_roles_granting_privileges:
+
+Granting privileges to a user
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To assign a role to a user, use the :ref:`credentials.users.\<username\>.roles <configuration_reference_credentials_users_name_roles>` option.
+In this example, the ``dbadmin`` user gets privileges granted to the ``super`` built-in role:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/credentials/config.yaml
+    :language: yaml
+    :start-at: credentials:
+    :end-at: [ super ]
+    :dedent:
+
+To create a new role, define it in the :ref:`credentials.roles.* <configuration_reference_credentials_roles_options>` section.
+In the example below, the ``writers_space_reader`` role gets privileges to select data in the ``writers`` space:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/credentials/config.yaml
+    :language: yaml
+    :start-after: spaces: [ books ]
+    :end-at: spaces: [ writers ]
+    :dedent:
+
+Then, you can assign this role to a user using :ref:`credentials.users.\<username\>.roles <configuration_reference_credentials_users_name_roles>` (``sampleuser`` in the example below):
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/credentials/config.yaml
+    :language: yaml
+    :start-at: sampleuser:
+    :end-at: [ writers_space_reader ]
+    :dedent:
+
+Apart from assigning a role to a user, you can grant specific privileges directly using :ref:`credentials.users.\<username\>.privileges <configuration_reference_credentials_users_name_privileges>`.
```

Comment:
Nit: intro part `Apart from assigning a role to a user` doesn't seem to add meaning, so I'd drop it to keep short. Up to you.

Reply:
Idea: maybe make separate headings `Granting roles` and `Granting privileges` to make the right ToC more explicit?